libfwupdplugin/fu-fmap-firmware.c

```diff
@@ -21,153 +21,34 @@ typedef struct {
 G_DEFINE_TYPE_WITH_PRIVATE (FuFmapFirmware, fu_fmap_firmware, FU_TYPE_FIRMWARE)
 #define GET_PRIVATE(o) (fu_fmap_firmware_get_instance_private (o))
 
-/* returns size of fmap data structure if successful, <0 to indicate error */
-static gint
-fmap_size (FuFmap *fmap)
-{
-	if (fmap == NULL)
-		return -1;
-
-	return sizeof (*fmap) + (fmap->nareas * sizeof (FuFmapArea));
-}
-
-/* brute force linear search */
 static gboolean
-fmap_lsearch (const guint8 *image, gsize len, gsize *offset, GError **error)
+fu_fmap_firmware_find_signature (const guint8 *buf, gsize bufsz,
+				 gsize *offset, GError **error)
 {
-	gsize i;
-	gboolean fmap_found = FALSE;
-
-	if (offset == NULL) {
-		g_set_error_literal (error,
-				     G_IO_ERROR,
-				     G_IO_ERROR_INVALID_DATA,
-				     "offset return not valid");
-		return FALSE;
-	}
+#ifdef HAVE_MEMMEM
```

Comment:
probably need @bleungatchromium to comment if this is OK in chromeos to depend on that.

Reply:
Alright. Let me double check both architectures we build to make sure we have this.

Reply:
It's really just depending on _GNU_SOURCE and glibc. Thanks!